src/confighttp.cpp

```diff
@@ -467,67 +611,63 @@ namespace confighttp {
     }
     catch (std::exception &e) {
       BOOST_LOG(warning) << "SaveApp: "sv << e.what();
-
-      outputTree.put("status", "false");
-      outputTree.put("error", "Invalid Input JSON");
+      bad_request(response, request, e.what());
       return;
     }
 
-    outputTree.put("status", "true");
+    outputTree.put("status", true);
     proc::refresh(config::stream.file_apps);
   }
 
   /**
    * @brief Delete an application.
    * @param response The HTTP response object.
    * @param request The HTTP request object.
+   *
+   * @api_examples{/api/apps/9999| DELETE| null}
    */
   void
   deleteApp(resp_https_t response, req_https_t request) {
     if (!authenticate(response, request)) return;
 
     print_req(request);
 
-    pt::ptree outputTree;
-    auto g = util::fail_guard([&]() {
-      std::ostringstream data;
+    int index;
 
-      pt::write_json(data, outputTree);
-      response->write(data.str());
-    });
-    pt::ptree fileTree;
+    pt::ptree outputTree;
     try {
+      pt::ptree fileTree;
+      pt::ptree newApps;
       pt::read_json(config::stream.file_apps, fileTree);
       auto &apps_node = fileTree.get_child("apps"s);
-      int index = stoi(request->path_match[1]);
+      index = stoi(request->path_match[1]);
 
-      if (index < 0) {
-        outputTree.put("status", "false");
-        outputTree.put("error", "Invalid Index");
+      if (std::clamp(index, 0, static_cast<int>(apps_node.size()) - 1) != index) {
+        bad_request(response, request, "index out of range, max index is "s + std::to_string(apps_node.size() - 1));
         return;
       }
-      else {
-        // Unfortunately Boost PT does not allow to directly edit the array, copy should do the trick
-        pt::ptree newApps;
-        int i = 0;
-        for (const auto &[k, v] : apps_node) {
-          if (i++ != index) {
-            newApps.push_back(std::make_pair("", v));
-          }
+
+      // Unfortunately Boost PT does not allow to directly edit the array, copy should do the trick
+      int i = 0;
+      for (const auto &[k, v] : apps_node) {
+        if (i++ != index) {
+          newApps.push_back(std::make_pair("", v));
         }
-        fileTree.erase("apps");
-        fileTree.push_back(std::make_pair("apps", newApps));
       }
+      fileTree.erase("apps");
+      fileTree.push_back(std::make_pair("apps", newApps));
+
       pt::write_json(config::stream.file_apps, fileTree);
     }
     catch (std::exception &e) {
       BOOST_LOG(warning) << "DeleteApp: "sv << e.what();
-      outputTree.put("status", "false");
-      outputTree.put("error", "Invalid File JSON");
+      bad_request(response, request, e.what());
       return;
     }
 
-    outputTree.put("status", "true");
```

Comment:
This exclusion from the try...catch block seems strange. Could we move it at the end of the try block? Then in the catch block we would need to return.

This applies everywhere and also is just a personal preference, so this point can be ignored.

Reply:
Yes, although I didn't because I didn't think it would be likely/possible to cause an exception. I am used to more explicit try/catches instead of these huge ones though.